lib/tackle/consumer/executor.ex

```diff
@@ -21,47 +21,104 @@ defmodule Tackle.Consumer.Executor do
     state =
       options
       |> Tackle.Consumer.State.configure!()
-      |> setup!()
 
-    {:ok, state}
+    {:ok, state, {:continue, :try_setup}}
   end
 
-  defp setup!(%{topology: topology} = state) do
-    {:ok, channel} =
-      setup_connection_channel(
-        state.connection_id,
-        state.rabbitmq_url,
-        state.prefetch_count
-      )
+  # Try to open channel and setup topology
+  def handle_continue(:try_setup, %{channel: channel, consumer_tag: nil} = state)
+      when not is_nil(channel) do
+    {:noreply, state, {:continue, :try_consume}}
+  end
 
-    Tackle.Consumer.Topology.setup!(channel, topology)
+  def handle_continue(:try_setup, %{channel: channel} = state) when not is_nil(channel) do
+    {:noreply, state}
+  end
 
-    # start actual consuming
-    {:ok, _consumer_tag} = AMQP.Basic.consume(channel, topology.queue)
+  def handle_continue(:try_setup, %{channel_retry_ref: ref} = state) when is_reference(ref) do
+    {:noreply, state}
+  end
 
-    # Sleep is needed for the ConnectionErrorsTest
-    Process.sleep(2)
+  def handle_continue(
+        :try_setup,
+        %{
+          connection_id: connection_name,
+          rabbitmq_url: rabbitmq_url,
+          prefetch_count: prefetch_count,
+          topology: topology,
+          reconnect_interval: reconnect_interval,
+          handler: handler_module
+        } = state
+      ) do
+    with {:ok, connection} <- Tackle.Connection.open(connection_name, rabbitmq_url),
+         {:ok, channel} <- Tackle.Channel.create(connection, prefetch_count) do
+      # Get notifications when the connection goes down
+      Process.monitor(channel.pid)
+      Tackle.Consumer.Topology.setup!(channel, topology)
 
-    Tackle.Consumer.State.started(state, channel: channel)
+      {:noreply, %{state | channel: channel, channel_retry_ref: nil}, {:continue, :try_consume}}
+    else
+      {:error, _} = error ->
+        Logger.error("#{handler_module} failed to setup channel due to `#{inspect(error)}`")
+        ref = Process.send_after(self(), :setup_retry_timeout, reconnect_interval)
```

Comment:
This is meant as a retry of the setup step after a timeout, right? If so, then wouldn't you agree that `:delayed_retry_setup` or something similar might be a better name to convey this meaning?